### PR TITLE
Add elastic-agent winlog input documentation

### PR DESCRIPTION
**Pull request by leehinman**

### CODE REVIEW

**Comment by leehinman** [comment]
@urso FYI this is the winlog documentation I'm proposing for agent.  I'm not sure if we want `index` in the list or not, I'm thinking it is easier to talk about if it is there and really easy to delete if we don't want it.

**Comment by urso** [comment]
@dedemorton How to we handled the `index` setting for the other inputs.

In general I would say that `index` is a low level setting on the Beats level, which is actually set by the agent when the data stream is configured. At the agent standalone level or Fleet this setting should not exist.

File: docs/en/ingest-management/elastic-agent/configuration/inputs/input-winlog.asciidoc [hunk 1]
```diff
@@ -0,0 +1,372 @@
+:input-type: winlog
+
+:vista_and_newer: This option is only available on operating systems +
+  supporting the Windows Event Log API (Microsoft Windows Vista and newer).
```
Comment (urso) [inline docs/en/ingest-management/elastic-agent/configuration/inputs/input-winlog.asciidoc]: Uhm, do we really still support Vista officially? For 7.x we might still support windows 7 for a while (inofficially), but if I'm not mistaken Vista support has been removed from Go maybe 2 or 3 years ago already.

File: docs/en/ingest-management/elastic-agent/configuration/inputs/input-winlog.asciidoc [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+Configure the `winlog` input to read messages from Microsoft Windows
+Event Channels.
+
+To configure this input, specify a `name` of the Microsof Windows
+Event Channel to subscribe to.
+
+You can get a list of available event logs by running `Get-WinEvent
+-ListLog` in PowerShell.  Most channels are tied to specific event
+publishers. Here is an example showing how to list all channels using
+PowerShell.
```
Comment (urso) [inline docs/en/ingest-management/elastic-agent/configuration/inputs/input-winlog.asciidoc:19]: Not sure we have had it in the old docs, but using 'Event Viewer' allows users to inspect actual channels before configuring them. Maybe nice to have for "newcomers" that don't have much experience with windows event logs.

File: docs/en/ingest-management/elastic-agent/configuration/inputs/input-winlog.asciidoc [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+LogName : Microsoft-Management-UI/Admin
+LogName : Microsoft-Rdms-UI/Admin
+LogName : Microsoft-Rdms-UI/Operational
+LogName : Microsoft-Windows-Windows Firewall With Advanced Security/Firewall
+...
+--------------------------------------------------------------------------------
+
+You must specify the full name of the channel in the configuration
+file.  To read events from an archived `.evtx` file you can specify
+the `name` as the absolute path (it cannot be relative) to the file.
```
Comment (urso) [inline docs/en/ingest-management/elastic-agent/configuration/inputs/input-winlog.asciidoc]: What does it take to read them? @kvch is working on adding 'parsers' to the new filestream input. Plus the new code structure might allow us to be more flexible in how we set up harvesters and prospectors (allow alternative implementations in the future). 

Although reading `*evtx` is still supported by this input, I wonder if we should not document that feature and think if/how we to change how this file format is supported in the future. Using the filesteam input we might combine these features such that users can configure a directory they can drop evtx files into and we would just pick those up and process them like regular log files. I feel with fleet, setups might be a little more rigid (due to users having to update configurations in Kibana) and we should consider how we can let users create a little more flexible configurations upfront.

File: docs/en/ingest-management/elastic-agent/configuration/inputs/input-winlog.asciidoc [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+
+// end::provider-setting
+
+include::input-shared-settings.asciidoc[tag=tags-setting]
+
+
+|===
+
+[[input-winlog-experimental-settings]]
+=== Experimental settings
```
Comment (urso) [inline docs/en/ingest-management/elastic-agent/configuration/inputs/input-winlog.asciidoc]: For how long are these experimental? Time to move them to Beta/GA?

**Comment by dedemorton** [comment]
> In general I would say that index is a low level setting on the Beats level, which is actually set by the agent when the data stream is configured. At the agent standalone level or Fleet this setting should not exist.

I agree. I've copied the setting over to the [PR for Filebeat inputs](https://github.com/elastic/observability-docs/pull/424), but I flagged it as something we should remove. We should only expose low-level details when absolutely necessary.

File: docs/en/ingest-management/elastic-agent/configuration/inputs/input-winlog.asciidoc (hunk 4 above)
Comment (leehinman) [reply docs/en/ingest-management/elastic-agent/configuration/inputs/input-winlog.asciidoc]: that I don't know.  @andrewkroh do you know?

**Comment by urso** [comment]
LGTM

File: docs/en/ingest-management/elastic-agent/configuration/inputs/input-winlog.asciidoc (hunk 4 above)
Comment (leehinman) [reply docs/en/ingest-management/elastic-agent/configuration/inputs/input-winlog.asciidoc]: I double checked with @andrewkroh and we can move them to production.  I'll open a PR to do the same in the winlogbeat docs.

**Comment by dedemorton** [comment]
For various reasons, https://github.com/elastic/observability-docs/pull/424 has been stalled for quite awhile. I am making some pretty dramatic changes to the format of the docs, and those changes will affect the winlogbeat additions. There are still a lot of open issues wrt how these settings get passed to Elastic Agent. I am just trying to unblock the main PR so we can get the discussion going again.

**Comment by dedemorton** [comment]
Closed because this is quite old. I no longer work on agent docs, and it looks like that team is taking a different approach (not documenting all the input settings and instead pointing to the Beats docs).

**Pull request closed without merging**